internal/runners/uploadingredient/uploadingredient.go

```diff
@@ -0,0 +1,343 @@
+package uploadingredient
+
+import (
+	"net/http"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ActiveState/cli/internal/captain"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/gqlclient"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/prompt"
+	"github.com/ActiveState/cli/internal/rtutils/p"
+	"github.com/ActiveState/cli/pkg/platform/api"
+	graphModel "github.com/ActiveState/cli/pkg/platform/api/graphql/model"
+	"github.com/ActiveState/cli/pkg/platform/api/graphql/request"
+	auth "github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/machinebox/graphql"
+	"github.com/skratchdot/open-golang/open"
+	"gopkg.in/yaml.v3"
+)
+
+type Params struct {
+	Name         string
+	Version      string
+	Namespace    string
+	Description  string
+	Authors      captain.UsersFlag
+	Depends      captain.PackagesFlag
+	Filepath     string
+	MetaFilepath string
+	Edit         bool
+	Editor       bool
+}
+
+type Runner struct {
+	auth    *auth.Auth
+	out     output.Outputer
+	prompt  prompt.Prompter
+	project *project.Project
+	client  *gqlclient.Client
+}
+
+type primeable interface {
+	primer.Outputer
+	primer.Auther
+	primer.Projecter
+	primer.Prompter
+}
+
+func New(prime primeable) *Runner {
+	client := gqlclient.NewWithOpts(
+		api.GetServiceURL(api.ServiceBuildPlan).String(), 0,
+		graphql.WithHTTPClient(http.DefaultClient),
+		graphql.UseMultipartForm(),
+	)
```

Comment:
I'm a bit surprised we're not passing around the gql client from higher in the stack into runners. If we are, why is there special setup here?

Reply:
Why hold off on structuring the BP client into the application?

Reply:
You mean in terms of future sprints? Fair; I've opened a ticket: https://activestatef.atlassian.net/browse/DX-1907

In terms of this ticket it feels significantly out of scope.